src/ripple/protocol/ErrorCodes.h

```diff
@@ -92,7 +92,7 @@ enum error_code_i
     rpcALREADY_MULTISIG      = 36,
     rpcALREADY_SINGLE_SIG    = 37,
     // unused                  38,
-    // unused                  39,
+    rpcBAD_KEY_TYPE          = 39,
```

Comment:
I'm not sure that's kosher. See this comment on line 38:
```
// Please only append to this table. Do not "fill-in" gaps and do not re-use
// or repurpose error code values.
```

Reply:
It's a question regarding the harm caused.  Presumably code `39` meant something in the past and is now unused.  If someone in the past wrote code that cared about the code `39` that part of their code has not been exercised in a while.  Now this re-use of `39` appears.  It's hard to say how their code would react.

The advice given in the comment on line 38 is the safest, most cautious, approach.

Reply:
I tried to go back in time and look at the history; I think using 39 is safe in this instance. But I’m fine with using a “fresh” error code, if we so desire.

I’ll point out (following up on previous discussion) that the numeric value should not be canonical; it’s just an artifact, really. The string key should be what’s used by client apps.

Reply:
I agree that the error code values should not be canonical.  But after I did a refactor that compressed the codes (changing lots of these values) we found out that change broke at least one partner, so we backed out that change.  Hyrum's law for the win: https://www.hyrumslaw.com

Or, in other words, the only way to get clients to not use a value is to not make it visible.  Anywhere.  Or any time.  There are no take-backs.